### PR TITLE
feat: walletconnect v2

### PR DESCRIPTION
**Pull request by intergalacticspacehighway**
# Why
PR to migrate to Wallet Connect v2

# TODO

- [x] Migrate ethers to viem
- [x] Upgrade rainbowkit
- [x] Upgrade wagmi
- [x] Migrate wallet connect clients on native

### CODE REVIEW

File: apps/expo/package.json [hunk 1]
```diff
@@ -67,7 +67,9 @@
     "@stripe/stripe-js": "^1.52.1",
     "@tradle/react-native-http": "^2.0.1",
     "@types/tinycolor2": "^1.4.3",
+    "@walletconnect/react-native-compat": "^2.7.8",
     "@walletconnect/react-native-dapp": "^1.8.0",
+    "@web3modal/react-native": "^1.0.0-alpha.8",
```
Comment (alantoa) [inline apps/expo/package.json]: I saw that this lib has been deprecated, do we still need it?
CleanShot 2023-06-26 at 8 55 22

Comment (hirbod) [reply apps/expo/package.json]: In Slack we received:

* @walletconnect/ethereum-provider:  2.8.3 or higher
* @walletconnect/modal: 2.5.4 or higher

@intergalacticspacehighway

Comment (intergalacticspacehighway) [reply apps/expo/package.json]: good catch. Let me remove this dep. We've forked it and using a custom version. Chatting with wallet connect team if they can add the features we need.

**Review by hirbod — APPROVED** [review]
LGTM

**Pull request merged**